dbm-ui/backend/db_services/mysql/resources/tendbcluster/query.py

```diff
@@ -109,31 +109,13 @@ def _to_cluster_representation(
         **kwargs,
     ) -> Dict[str, Any]:
         """将集群对象转为可序列化的 dict 结构"""
-
-        def get_remote_infos(insts: List[StorageInstance]):
-            """获取remote信息，并补充分片信息"""
-            remote_infos = defaultdict(list)
-            for inst in insts:
-                try:
-                    related = "as_ejector" if inst.instance_inner_role == InstanceInnerRole.MASTER else "as_receiver"
-                    shard_id = getattr(inst, related).all()[0].tendbclusterstorageset.shard_id
-                except Exception:
-                    # 如果无法找到shard_id，则默认为-1。有可能实例处于restoring状态(比如集群容量变更时)
-                    shard_id = -1
-
-                remote_infos[inst.instance_inner_role].append({**inst.simple_desc, "shard_id": shard_id})
-
-            remote_infos[InstanceInnerRole.MASTER.value].sort(key=lambda x: x.get("shard_id", -1))
-            remote_infos[InstanceInnerRole.SLAVE.value].sort(key=lambda x: x.get("shard_id", -1))
-            return remote_infos[InstanceInnerRole.MASTER.value], remote_infos[InstanceInnerRole.SLAVE.value]
-
         # 获取spider，remote角色实例信息
         spider = {
             role: [inst.simple_desc for inst in cluster.proxies if inst.tendbclusterspiderext.spider_role == role]
             for role in TenDBClusterSpiderRole.get_values()
         }
 
-        remote_db, remote_dr = get_remote_infos(cluster.storages)
+        remote_db, remote_dr = TenDBClusterClusterHandler.get_remote_infos(cluster.storages)
```

Comment:
remote_db = {**inst.simple_desc, "shard_id": shard_id for inst in remote_db}
remote_dr = {**inst.simple_desc, "shard_id": shard_id for inst in remote_dr}